.changeset/ninety-onions-bow.md

```diff
@@ -0,0 +1,5 @@
+---
+'create-astro': minor
+---
+
+Automatically installs the required dependencies to run the `astro check` command when TypeScript mode is selected.
```

Comment:
Fair. I think that final suggestion is the clearest! One of these days we'll all agree on how the CLI should handle this 😅

```suggestion
Automatically installs the required dependencies to run the astro check command when the user indicates they plan to write TypeScript.
```